src/main/java/com/google/sps/ProductDetectionAPIImpl.java

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageRequest.Builder;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.ColorInfo;
+import com.google.cloud.vision.v1.DominantColorsAnnotation;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageSource;
+
+import com.google.protobuf.ByteString;
+
+import com.google.sps.data.ProductDetectionData;
+
+import java.awt.Color;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Uses Cloud Vision API to detect product photo content.
+ */
+public class ProductDetectionAPIImpl implements ProductDetectionAPI {
+
+  public ProductDetectionData detectProductPhotoContent(byte[] imageBytes) 
+      throws PhotoDetectionException {
+    // Build Image object by converting bytes array.
+    ByteString byteString = ByteString.copyFrom(imageBytes);
+    Image image = Image.newBuilder().setContent(byteString).build();
+
+    // Build the image annotation request.
+    // {@code requestBuilder} holds the image and the features to detect from it.
+    Builder requestBuilder = AnnotateImageRequest.newBuilder();
+
+    Feature labelDetectionFeature = 
+        Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
+    requestBuilder.addFeatures(labelDetectionFeature);
+
+    Feature logoDetectionFeature =
+        Feature.newBuilder().setType(Feature.Type.LOGO_DETECTION).build();
+    requestBuilder.addFeatures(logoDetectionFeature);
+
+    Feature imagePropertiesDetectionFeature =
+        Feature.newBuilder().setType(Feature.Type.IMAGE_PROPERTIES).setMaxResults(5).build();
+    requestBuilder.addFeatures(imagePropertiesDetectionFeature);
```

Comment:
I think we can keep these features in a Constant.java class like Changcheng suggested and I did in my pr.
So that we can be sure that we are using the right and updated constants.

Reply:
Here I would like to ask for @ChangchengXiao 's opinion. So what I'm thinking is that it's easier for the reader to see the initialization of the features in the method that does the detection. Also, these variables are used only here, so this also adds to my resoning. I do not have a problem with making them constants, but I keep thinking about what I just said... please tell me if my reasoning is wrong :D

Reply:
So I was using them in my tests as well so it was better to keep them in a separate class as constants.

Reply:
AFAIK, we shall not test the actual Could Vision API behaviour :) so there's no need for having the cloud vision feature object in the tests

Reply:
Yeh, I was kind of in the middle of figuring out how to do the testing part at that time 😅
I see now, what you did looks appropriate 😄